collectors/raw_request.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+class QM_Collector_Raw_Request extends QM_Collector {
+
+	public $id = 'raw_request';
+
+	/**
+	 * Collector name.
+	 *
+	 * This is unused.
+	 *
+	 * @return string
+	 */
+	public function name() {
+		return __( 'Request Data', 'query-monitor' );
+	}
+
+	/**
+	 * Extracts headers from a PHP-style $_SERVER array.
+	 *
+	 * From WP_REST_Server::get_headers()
+	 *
+	 * @param array $server Associative array similar to `$_SERVER`.
+	 * @return array Headers extracted from the input.
+	 */
+	protected function get_headers( $server ) {
```

Comment:
I copied this across, but `rest_get_server()->get_headers()` is equivalent; it forces the API to load though, so not the greatest.